quarkus-extension/deployment/src/main/java/io/apicurio/openapi/server/generator/deployment/codegen/ApicurioOpenApiServerCodegen.java

```diff
@@ -39,29 +44,53 @@ public boolean shouldRun(Path sourceDir, Config config) {
-
-                try (Stream<Path> openApiFilesPaths = Files.walk(openApiDir)) {
-                    openApiFilesPaths
-                            .filter(Files::isRegularFile)
-                            .map(Path::toString)
-                            .filter(s -> s.endsWith(this.inputExtension()))
-                            .map(Path::of).forEach(openApiResource -> {
-                                if (openApiResource.toFile().getName().equals(context.config().getOptionalValue(CodegenConfig.getSpecPropertyName(), String.class).get())) {
-                                try {
-                                    apicurioCodegenWrapper.generate(openApiResource);
-                                } catch (CodeGenException e) {
-                                    e.printStackTrace();
-                                }
-                            }});
+        final Optional<String> spec = context.config().getOptionalValue(CodegenConfig.getSpecPropertyName(), String.class);
 
+        if (Files.isDirectory(openApiDir) && spec.isPresent()) {
+            final ApicurioCodegenWrapper apicurioCodegenWrapper =
+                    new ApicurioCodegenWrapper(outDir.toFile(), projectSettings(context.config(), context));
+
+            try (Stream<Path> openApiFilesPaths = Files.walk(openApiDir)) {
+                openApiFilesPaths
+                    .filter(Files::isRegularFile)
+                    .filter(path -> path.toString().endsWith(inputExtension()))
+                    .filter(path -> path.toFile().getName().endsWith(spec.get()))
+                    .forEach(openApiResource -> {
+                        try {
+                            apicurioCodegenWrapper.generate(openApiResource);
+                        } catch (CodeGenException e) {
+                            e.printStackTrace();
+                        }
+                    });
             } catch (IOException e) {
                 throw new CodeGenException("Failed to generate java files from OpenApi file in " + openApiDir.toAbsolutePath(), e);
             }
             return true;
         }
+
         return false;
     }
+
+    private static JaxRsProjectSettings projectSettings(Config config, CodeGenContext context) {
+        JaxRsProjectSettings projectSettings = new JaxRsProjectSettings();
+
+        projectSettings.setJavaPackage(config
+                .getOptionalValue(getBasePackagePropertyName(), String.class)
+                .orElse(DEFAULT_PACKAGE));
+
+        projectSettings.setReactive(config
+                .getOptionalValue(CodegenConfig.getReactivePropertyName(), Boolean.class)
+                .orElseGet(() -> context.applicationModel()
+                            .getDependencies()
+                            .stream()
+                            .filter(dependency -> "quarkus-resteasy-reactive".equals(dependency.getArtifactId()))
+                            .anyMatch(dependency -> "io.quarkus".equals(dependency.getGroupId()))));
```

Comment:
This feels a bit brittle. I'm not finding anything better available to `CodeGenProvider` implementations. If others agree, I'll keep digging to see if there is a better approach.

Reply:
shrug emoji  :)

Reply:
Ok - it turns out there is a collection of capabilities in the application model that is better than checking Maven coordinates. I've updated the PR to use that instead.